app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,53 @@
+class ReviewsController < ApplicationController
+  before_action :authenticate_user!
+
+  def new
+    @reviews = Review.all
+  end
+
+  def create
+    @review = Review.new(review_params)
+    @dinosaur = Dinosaur.find(params[:dinosaur_id])
+    @review.dinosaur = @dinosaur
+
+    if @review.save
+      flash[:notice] = "Review added!"
+      redirect_to dinosaur_path(@dinosaur)
+    else
+      flash[:errors] = @review.errors.full_messages.join(". ")
+      render :'dinosaurs/show'
+    end
+  end
+
+  def edit
+    @review = Review.find(params[:id])
+  end
+
+  def update
+    @review = Review.find(params[:id])
+    if @review.update_attributes(review_params)
+      flash[:notice] = "Review Updated!"
+      redirect_to dinosaur_path(@review.dinosaur_id)
+    else
+      flash[:errors] = @review.errors.full_messages.join('. ')
+      render 'dinosaurs/show'
+    end
+  end
+
+  def destroy
+    # @dinosaur = Dinosaur.find(params[:id])
+    @review = Review.find(params[:id])
+    @review.destroy
+    flash[:notice] = "Review deleted!"
+    redirect_to dinosaur_path(@review.dinosaur_id)
+  end
+
+  private
+
+  def review_params
+    params.require(:review).permit(
+    :body, 
```

Comment:
Align the parameters of a method call if they span more than one line.<br>Indent the first parameter one step more than the previous line.<br>Trailing whitespace detected.